pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -3693,6 +3698,134 @@ func (s *PluginSuite) TestDeleteBundleDissociateRegistrationEntries() {
+	s.Require().NoError(err)
+	s.Require().Equal(expectedEntryIDs, resp.EntryIDs)
+
+	// Delete second entry
+	s.deleteRegistrationEntry(entry2.EntryId)
+	expectedEntryIDs = append(expectedEntryIDs, entry2.EntryId)
+
+	resp, err = s.ds.ListRegistrationEntriesEvents(ctx, &datastore.ListRegistrationEntriesEventsRequest{})
+	s.Require().NoError(err)
+	s.Require().Equal(expectedEntryIDs, resp.EntryIDs)
+
+	// Check filtering events by id
+	tests := []struct {
+		name                 string
+		greaterThanEventID   uint
+		expectedEntryIDs     []string
+		expectedFirstEntryID uint
+	}{
+		{
+			name:                 "All Events",
+			greaterThanEventID:   0,
+			expectedFirstEntryID: 1,
+			expectedEntryIDs:     []string{entry1.EntryId, entry2.EntryId, entry1.EntryId, entry2.EntryId},
+		},
+		{
+			name:                 "Half of the Events",
+			greaterThanEventID:   2,
+			expectedFirstEntryID: 3,
+			expectedEntryIDs:     []string{entry1.EntryId, entry2.EntryId},
+		},
+		{
+			name:                 "None of the  Events",
+			greaterThanEventID:   4,
+			expectedFirstEntryID: 0,
+			expectedEntryIDs:     nil,
+		},
+	}
+	for _, test := range tests {
+		s.T().Run(test.name, func(t *testing.T) {
+			resp, err = s.ds.ListRegistrationEntriesEvents(ctx, &datastore.ListRegistrationEntriesEventsRequest{
+				GreaterThanEventID: test.greaterThanEventID,
+			})
+			s.Require().NoError(err)
+			s.Require().Equal(test.expectedFirstEntryID, resp.FirstEventID)
+			s.Require().Equal(test.expectedEntryIDs, resp.EntryIDs)
+		})
+	}
+}
+
+func (s *PluginSuite) TestPruneRegistrationEntriesEvents() {
+	entry := &common.RegistrationEntry{
+		Selectors: []*common.Selector{
+			{Type: "Type1", Value: "Value1"},
+		},
+		SpiffeId: "SpiffeId",
+		ParentId: "ParentId",
+	}
+
+	createdRegistrationEntry := s.createRegistrationEntry(entry)
+	resp, err := s.ds.ListRegistrationEntriesEvents(ctx, &datastore.ListRegistrationEntriesEventsRequest{})
+	s.Require().NoError(err)
+	s.Require().Equal(createdRegistrationEntry.EntryId, resp.EntryIDs[0])
+
+	for _, tt := range []struct {
+		name             string
+		olderThan        time.Duration
+		expectedEntryIDs []string
+	}{
+		{
+			name:             "Don't prune valid events",
+			olderThan:        1 * time.Hour,
+			expectedEntryIDs: []string{createdRegistrationEntry.EntryId},
+		},
+		{
+			name:             "Prune old events",
+			olderThan:        0 * time.Second,
+			expectedEntryIDs: nil,
+		},
+	} {
+		s.T().Run(tt.name, func(t *testing.T) {
+			s.Require().Eventuallyf(func() bool {
+				err = s.ds.PruneRegistrationEntriesEvents(ctx, tt.olderThan)
+				s.Require().NoError(err)
+				resp, err := s.ds.ListRegistrationEntriesEvents(ctx, &datastore.ListRegistrationEntriesEventsRequest{})
+				s.Require().NoError(err)
+				return reflect.DeepEqual(tt.expectedEntryIDs, resp.EntryIDs)
+			}, 10*time.Second, 50*time.Millisecond, "Failed to prune entries correctly")
```

Comment:
what do you think about adding clock to ds layer (and a mock here), and use that instead of time.Now.
That will allow you to add test cases where you can advance time and have more robust test cases where you have more events and prune partially

Reply:
I think thats a great idea. Is it ok to that in a follow on pr? I think its a little out of scope for this one.